### ISSUE TITLE
Unhandled exception when running the AppImage

### ISSUE THREAD

**Issue by probonopd** [h3poteto/whalebird-desktop#1163]
Getting an unhandled exception:

```
[22:22:52.905] [error] Error: ENOENT: no such file or directory, open '/home/travis/.config/Whalebird/db/unread_notification.db'

[Error: ENOENT: no such file or directory, open '/home/travis/.config/Whalebird/cache/hashtag.db'] {

  errno: -2,

  code: 'ENOENT',

  syscall: 'open',

  path: '/home/travis/.config/Whalebird/cache/hashtag.db'

}

[22:22:52.969] [error] Error: empty

    at new n (/run/firejail/appimage/.appimage-9127/resources/app.asar/dist/electron/main.js:1:25757)

    at /run/firejail/appimage/.appimage-9127/resources/app.asar/dist/electron/main.js:1:21401

    at newArguments.<computed> (/run/firejail/appimage/.appimage-9127/resources/app.asar/node_modules/nedb/lib/executor.js:29:17)

    at Cursor.execFn (/run/firejail/appimage/.appimage-9127/resources/app.asar/node_modules/nedb/lib/datastore.js:484:12)

    at callback (/run/firejail/appimage/.appimage-9127/resources/app.asar/node_modules/nedb/lib/cursor.js:126:19)

    at /run/firejail/appimage/.appimage-9127/resources/app.asar/node_modules/nedb/lib/cursor.js:193:12

    at /run/firejail/appimage/.appimage-9127/resources/app.asar/node_modules/nedb/lib/datastore.js:329:14

    at Object.async.eachSeries (/run/firejail/appimage/.appimage-9127/resources/app.asar/node_modules/nedb/node_modules/async/lib/async.js:130:20)

    at /run/firejail/appimage/.appimage-9127/resources/app.asar/node_modules/nedb/lib/datastore.js:323:11

    at fn (/run/firejail/appimage/.appimage-9127/resources/app.asar/node_modules/nedb/node_modules/async/lib/async.js:582:34)

[22:22:53.197] [info] System proxy configuration: DIRECT

(whalebird:5): libappindicator-WARNING **: Unable to get the session bus: Unknown or unsupported transport 'disabled' for address 'disabled:'

(whalebird:5): LIBDBUSMENU-GLIB-WARNING **: Unable to get session bus: Unknown or unsupported transport 'disabled' for address 'disabled:'

(node:5) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'source' of undefined

    at e.<anonymous> (/run/firejail/appimage/.appimage-9127/resources/app.asar/dist/electron/main.js:1:40721)

    at r (/run/firejail/appimage/.appimage-9127/resources/app.asar/dist/electron/main.js:1:39648)

    at Object.next (/run/firejail/appimage/.appimage-9127/resources/app.asar/dist/electron/main.js:1:38952)

    at s (/run/firejail/appimage/.appimage-9127/resources/app.asar/dist/electron/main.js:1:38717)

(node:5) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'source' of undefined

    at e.<anonymous> (/run/firejail/appimage/.appimage-9127/resources/app.asar/dist/electron/main.js:1:40721)

    at r (/run/firejail/appimage/.appimage-9127/resources/app.asar/dist/electron/main.js:1:39648)

    at Object.next (/run/firejail/appimage/.appimage-9127/resources/app.asar/dist/electron/main.js:1:38952)

    at s (/run/firejail/appimage/.appimage-9127/resources/app.asar/dist/electron/main.js:1:38717)

(node:5) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)

(node:5) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)

(node:5) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

(node:5) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

[22:22:54.306] [error] Error: empty

    at new n (/run/firejail/appimage/.appimage-9127/resources/app.asar/dist/electron/main.js:1:25757)

    at /run/firejail/appimage/.appimage-9127/resources/app.asar/dist/electron/main.js:1:21401

    at newArguments.<computed> (/run/firejail/appimage/.appimage-9127/resources/app.asar/node_modules/nedb/lib/executor.js:29:17)

    at Cursor.execFn (/run/firejail/appimage/.appimage-9127/resources/app.asar/node_modules/nedb/lib/datastore.js:484:12)

    at callback (/run/firejail/appimage/.appimage-9127/resources/app.asar/node_modules/nedb/lib/cursor.js:126:19)

    at /run/firejail/appimage/.appimage-9127/resources/app.asar/node_modules/nedb/lib/cursor.js:193:12

    at /run/firejail/appimage/.appimage-9127/resources/app.asar/node_modules/nedb/lib/datastore.js:329:14

    at Object.async.eachSeries (/run/firejail/appimage/.appimage-9127/resources/app.asar/node_modules/nedb/node_modules/async/lib/async.js:130:20)

    at /run/firejail/appimage/.appimage-9127/resources/app.asar/node_modules/nedb/lib/datastore.js:323:11

    at fn (/run/firejail/appimage/.appimage-9127/resources/app.asar/node_modules/nedb/node_modules/async/lib/async.js:582:34)
```

What does this mean?

Reference:
https://github.com/AppImage/appimage.github.io/pull/1844

Full test log:
https://api.travis-ci.org/v3/job/622913644/log.txt

Test script:
https://github.com/AppImage/appimage.github.io/blob/master/code/worker.sh

**Comment by h3poteto** [comment]
`cache/hashtag.db` is for cache, so it is not required for operation.
`db/unread_notification.db` is setting file, but if does not exist, the default values are used. So don't mind these files.

I will investigate only for UnhandledPromiseRejection.

**Comment by h3poteto** [comment]
I fixed it.

@probonopd Could you try again?

**Comment by probonopd** [comment]
Happy to re-run the automated test once a new AppImage is available for download, but I still only see 3.0.0 available for download as an AppImage?

**Comment by h3poteto** [comment]
That's good.
I'm sorry I don't release next version, and the fix will be included in the next release. So please wait a little more.

**Comment by h3poteto** [comment]
I released v3.0.1, please use it.
https://github.com/h3poteto/whalebird-desktop/releases/tag/3.0.1
